momepy/preprocessing.py

```diff
@@ -740,3 +748,240 @@ def _get_extrapolated_line(coords, tolerance, point=False):
     if point:
         return b
     return pygeos.linestrings([a, b])
+
+def _selecting_rabs_from_poly(gdf, circom_threshold = 0.7, perc_area_threshold = 0.85, include_adjacent=True) :
+    """
+    From a GeoDataFrame of polygons, returns a GDF of polygons that are 
+    above the Circular Compaactness threshold.
+    
+    Return
+    ________
+    GeoDataFrames : round abouts and adjacent polygons
+    """
+    # calculate parameters
+    gdf["area"] = gdf.geometry.area
+    gdf["circom"] = CircularCompactness(gdf, "area").series
+    
+    #selecting round about polygons based on compactness
+    rab = gdf[gdf.circom > circom_threshold]
+    #exclude those above the area threshold
+    area_threshold = gdf.area.quantile(perc_area_threshold)
+    rab = rab[rab.area < area_threshold]
+    
+    if include_adjacent == True :
+        #selecting the adjacent areas that are of smaller than itself
+        rab_adj = gpd.sjoin(gdf, rab, predicate = 'intersects')
+        rab_adj = rab_adj[rab_adj.area_right >= rab_adj.area_left]
+        
+        adj_long_polys = []
+        #iterate through original rabs
+        for row in rab.itertuples() :
+            #creating a distance (diameter) for limiting the max dist of adjacent
+            minx, _ , maxx, _ = row.geometry.bounds 
+            rab_diameter = maxx - minx
+            
+            #creating the subgroup that could be affected
+            rab_group = rab_adj[rab_adj.index_right==row[0]]
+            for rrow in rab_group.itertuples():
+                hdist = row.geometry.hausdorff_distance(rrow.geometry)
+
+                if hdist > rab_diameter :
+                    adj_long_polys.append(rrow[0]) #if hausdorff_distance is larger than the diameter add to list
+           
+        rab_plus = rab_adj.drop(adj_long_polys)
+        
+    else:
+        rab['index_right'] = rab.index
+        rab_plus = rab
+    
+    rab_plus = rab_plus[['geometry', 'index_right']] #only keeping relevant fields
```

Comment:
You can't be certain you will always get "geometry" and "index_right" columns unless you sanitise the given gdf yourself. What a user may pass can have custom geometry columns name and a named index and this will no longer work.

Reply:
humm good point 👍 
I'm not sure how to deal with the 'index_right' just yet (possibly by reseting the index at the very begining?!)..
but for now I'll implement `geom_col = rab_plus.geometry.name` for the geometry column

Reply:
You can reset index but you should reconstruct it afterwards, to return gdf in a same structure as it comes in.